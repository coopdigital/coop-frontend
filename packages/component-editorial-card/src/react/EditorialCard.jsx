import React from "react";
import PropTypes from "prop-types";
// import '../css/EditorialCard.pcss';

const EditorialCard = ({ title, text, link, type, testId }) => {
  const tagAttributes = {
    className: "coop coop-c-editorialcard coop-c-editorialcard--horizontal",
    role: type === "warn" || type === "error" ? "alert" : "status",
  };

  return (
    <div
      data-testid={testId}
      className={tagAttributes.className}
      role={tagAttributes.role}
    >
      <div className="coop-c-editorialcard__inner ">
        <h3 className="coop-c-editorialcard__title coop-c-notification__heading">{title}</h3>
        <p className="coop-c-notification__p">
          {link ? (
            <a href={link} className="coop-c-notification__link">
              {text}
            </a>
          ) : (
            text
          )}
        </p>
      </div>
    </div>
  );
};

EditorialCard.defaultProps = {
  type: "info",
};

EditorialCard.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  link: PropTypes.string,
  type: PropTypes.oneOf(["info", "warn", "error", "success"]),
  testId: PropTypes.string,
};

export default EditorialCard;
