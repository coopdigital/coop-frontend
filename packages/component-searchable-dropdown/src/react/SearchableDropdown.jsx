import React, { useRef, useEffect } from "react";
import SearchableDropdownLibrary from "../lib/SearchableDropdownLibrary.js";

const SearchableDropdown = () => {
  const comboboxRef = useRef();

  useEffect(() => {
    new SearchableDropdownLibrary(comboboxRef.current).setup();
  }, []);

  return (
    <div ref={comboboxRef}>
      <div data-combobox>
        <label htmlFor="combobox">How you know them</label>
        <input
          name="combobox"
          className="coop-form__input"
          type="text"
          placeholder="Select option"
          data-input
          data-testid="combobox-input"
          role="combobox"
          aria-controls="comboboxvalues"
          aria-autocomplete="list"
          aria-expanded="false"
        />
        <div
          style={{ display: "none" }}
          data-dropdown
          data-testid="combobox-dropdown"
        >
          <ul id="comboboxvalues" role="listbox" data-list>
            {/* These valuse would be mapped from a data prop */}
            <li role="option" aria-selected="false" data-option>
              Partner
            </li>
            <li role="option" aria-selected="false" data-option>
              Spouse
            </li>
            <li role="option" aria-selected="false" data-option>
              Parent
            </li>
            <li role="option" aria-selected="false" data-option>
              Child
            </li>
            <li role="option" aria-selected="false" data-option>
              Brother
            </li>
            <li role="option" aria-selected="false" data-option>
              Grandchild
            </li>
            <li role="option" aria-selected="false" data-option>
              Father
            </li>
          </ul>
        </div>
      </div>
    </div>
  );
};

export default SearchableDropdown;
