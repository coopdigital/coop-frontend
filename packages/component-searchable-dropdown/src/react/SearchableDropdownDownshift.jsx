/* eslint-disable indent */
import React, { useState } from 'react';
import { useCombobox } from 'downshift';
import PropTypes from 'prop-types';
import { matchSorter } from 'match-sorter';

const SearchableDropdown = ({
  className,
  compact,
  id,
  label,
  noResults,
  onSelect,
  options,
  placeholder,
  style,
}) => {
  const [inputItems, setInputItems] = useState(options);
  const sizeVariant = compact ? 'compact' : '';
  const noResultsText = noResults || 'No relevant options';
  const placeholderValue = placeholder || null;

  const {
    isOpen,
    closeMenu,
    getToggleButtonProps,
    getLabelProps,
    getMenuProps,
    getInputProps,
    getComboboxProps,
    openMenu,
    getItemProps,
  } = useCombobox({
    id,
    items: inputItems,
    onInputValueChange: ({ inputValue }) => {
      setInputItems(matchSorter(options, inputValue));
    },
    onSelectedItemChange: ({ selectedItem: selected }) => {
      if (typeof onSelect === 'function') onSelect(selected);
      closeMenu();
    },
  });

  return (
    <div className={`coop-c-combobox ${sizeVariant} ${className}`} style={style}>
      <label {...getLabelProps()} htmlFor={`${id}-input`}>
        {label}
      </label>
      <div {...getComboboxProps()}>
        <input {...getInputProps({ onFocus: openMenu })} placeholder={placeholderValue} />
        <button type="button" {...getToggleButtonProps()} aria-label="toggle menu">
          <svg xmlns="http://www.w3.org/2000/svg" height="24" width="24">
            <path d="m12 15.375-6-6 1.4-1.4 4.6 4.6 4.6-4.6 1.4 1.4Z" />
          </svg>
        </button>
      </div>
      <ul {...getMenuProps()}>
        {isOpen && inputItems.length
          ? inputItems.map((item, index) => (
              <li {...getItemProps({ item, index })} key={`${item}${index}`}>
                {item}
              </li>
            ))
          : null}
        {isOpen && !inputItems.length ? (
          <li className="coop-c-combobox--null">{noResultsText}</li>
        ) : null}
      </ul>
    </div>
  );
};

export { SearchableDropdown };

SearchableDropdown.propTypes = {
  className: PropTypes.string,
  compact: PropTypes.bool,
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  noResults: PropTypes.string,
  onSelect: PropTypes.func,
  options: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string])).isRequired,
  placeholder: PropTypes.string,
  style: PropTypes.object,
};
