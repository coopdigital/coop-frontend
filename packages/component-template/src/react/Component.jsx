import React from "react";

const Component = () => <div class="coop-c-component">Example HTML</div>;

export default Component;
