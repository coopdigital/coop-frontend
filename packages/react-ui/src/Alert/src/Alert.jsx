import React from "react";
import PropTypes from "prop-types";
import classNames from "../../utils/classNames";

const Alert = ({ type, heading, link, children, ...props }) => {
  const { href, text } = link;

  const classes = classNames("coop-c-notification", [
    type === "warn" && "coop-c-notification--alert",
    type === "error" && "coop-c-notification--error",
  ]);

  const role = function (type) {
    if (type === "warn" || type == "error") {
      return "alert";
    } else {
      return "status";
    }
  };

  const tagAttributes = {
    className: classes,
    role: role(type),
    ...props,
  };

  return (
    <div {...tagAttributes}>
      <h3 className="coop-c-notification__heading">{heading}</h3>
      <p className="coop-c-notification__p">
        {href && (
          <a href={href} className="coop-c-notification__link">
            {text}
          </a>
        )}
      </p>
      {children && <p className="coop-c-notification__p">{children}</p>}
    </div>
  );
};

Alert.defaultProps = {
  type: "info",
  link: {
    href: "",
    text: "",
  },
  children: null,
};

Alert.propTypes = {
  type: PropTypes.oneOf(["info", "warn", "error"]),
  heading: PropTypes.string.isRequired,
  link: PropTypes.shape({
    href: PropTypes.string,
    text: PropTypes.string,
  }),
  children: PropTypes.node,
};

export default Alert;
