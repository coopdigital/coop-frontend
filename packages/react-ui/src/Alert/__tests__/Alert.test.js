/**
 * @jest-environment jsdom
 */

import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Alert from "../src/index";

afterEach(cleanup);

describe("Alert", () => {
  it("should render default Alert", () => {
    const { getByText } = render(<Alert heading="Alert heading" />);
    expect(getByText("Alert heading")).toBeInTheDocument;
  });

  it("should work with all variants", () => {
    const link = { href: "#", text: "Example message" };

    const variantMap = new Map([
      [
        "alert-warn",
        {
          class: "coop-c-notification--alert",
          heading: "Warning Alert",
          role: "alert",
        },
      ],
      [
        "alert-error",
        {
          class: "coop-c-notification--error",
          heading: "Error Alert",
          role: "alert",
        },
      ],
    ]);

    const Variants = () => (
      <div>
        <Alert type="info" heading="Info Alert" link={link} />
        <Alert
          data-testid="alert-warn"
          type="warn"
          heading="Warning Alert"
          link={link}
        />
        <Alert
          data-testid="alert-error"
          type="error"
          heading="Error Alert"
          link={link}
        />
        <Alert data-testid="alert-default" heading="Alert with no link">
          Standard message
        </Alert>
      </div>
    );
    /**
     * Default and alert type info are the same variant...
     * when passing no type prop the alert deafults to the info alert type.
     * Is it needed?
     */

    const { getByText, getByTestId } = render(<Variants />);

    variantMap.forEach((value, key) => {
      expect(getByTestId(key)).toHaveClass(value.class);

      //magic string for attribute??
      expect(getByTestId(key)).toHaveAttribute("role", value.role);
      expect(getByText(value.heading)).toBeInTheDocument();
    });
  });
});
