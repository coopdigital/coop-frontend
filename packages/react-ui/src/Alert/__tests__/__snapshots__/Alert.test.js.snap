// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should work with all variants 1`] = `
<DocumentFragment>
  <div>
    <div
      class="coop-c-notification"
      role="status"
    >
      <h3
        class="coop-c-notification__heading"
      >
        Info Alert
      </h3>
      <p
        class="coop-c-notification__p"
      >
        <a
          class="coop-c-notification__link"
          href="#"
        >
          Example message
        </a>
      </p>
    </div>
    <div
      class="coop-c-notification coop-c-notification--alert"
      data-testid="alert-warn"
      role="status"
    >
      <h3
        class="coop-c-notification__heading"
      >
        Warning Alert
      </h3>
      <p
        class="coop-c-notification__p"
      >
        <a
          class="coop-c-notification__link"
          href="#"
        >
          Example message
        </a>
      </p>
    </div>
    <div
      class="coop-c-notification coop-c-notification--error"
      data-testid="alert-error"
      role="status"
    >
      <h3
        class="coop-c-notification__heading"
      >
        Error Alert
      </h3>
      <p
        class="coop-c-notification__p"
      >
        <a
          class="coop-c-notification__link"
          href="#"
        >
          Example message
        </a>
      </p>
    </div>
    <div
      class="coop-c-notification"
      data-testid="alert-default"
      role="status"
    >
      <h3
        class="coop-c-notification__heading"
      >
        Alert with no link
      </h3>
      <p
        class="coop-c-notification__p"
      />
      <p
        class="coop-c-notification__p"
      >
        Standard message
      </p>
    </div>
  </div>
</DocumentFragment>
`;
